File: Daily Test/Java/Binary Reverse Pairs 4.12.2025.java
import java.util.*;

class Main {
    public static String toBinary(int n) {
        StringBuilder sb = new StringBuilder();
        while (n > 0) {
            sb.append(n % 2);
            n /= 2;
        }
        return sb.toString();  // reversed binary
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int[] arr = new int[n];
        for (int i = 0; i < n; i++)
            arr[i] = sc.nextInt();

        boolean found = false;

        for (int i = 0; i < n - 1; i++) {
            String bi = toBinary(arr[i]);

            for (int j = i + 1; j < n; j++) {
                String bj = toBinary(arr[j]);
                String revbj = new StringBuilder(bj).reverse().toString();

                if (bi.equals(revbj)) {
                    System.out.println(arr[i] + " " + arr[j]);
                    found = true;
                }
            }
        }

        if (!found)
            System.out.println(-1);
    }
}

File: Daily Test/C++/Binary Reverse Pairs 4.12.2025.cpp
#include <bits/stdc++.h>
using namespace std;

string toBinary(int n) {
    string s = "";
    while(n > 0) {
        s += char('0' + (n % 2));
        n /= 2;
    }
    return s; // reversed binary
}

int main() {
    int n;
    cin >> n;
    vector<int> arr(n);

    for(int i = 0; i < n; i++) cin >> arr[i];

    bool found = false;

    for(int i = 0; i < n - 1; i++) {
        string bi = toBinary(arr[i]);

        for(int j = i + 1; j < n; j++) {
            string bj = toBinary(arr[j]);
            string rj = bj;
            reverse(rj.begin(), rj.end());

            if(bi == rj) {
                cout << arr[i] << " " << arr[j] << "\n";
                found = true;
            }
        }
    }

    if(!found) cout << -1;

    return 0;
}